cli/pkg/timescaledb_secrets/create.go

```diff
@@ -0,0 +1,316 @@
+	if err != nil {
+		return err
+	}
+
+	if t.EnableS3Backup {
+		err = t.createS3BackupForTimescaleDB()
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("skipping to create TimescaleDB s3 backup secret as backup option is disabled.")
+	}
+
+	return nil
+}
+
+// randomPassword generates a random alphanumeric password of a given length.
+// Compatible with passwords generated by the postgres operator
+func randomPassword(n int) ([]byte, error) {
+	var passwordChars = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+	b := make([]byte, n)
+	_, err := crand.Read(b)
+	if err != nil {
+		return b, fmt.Errorf("failed to generate random password %v", err)
+	}
+	for i := 0; i < n; i++ {
+		b[i] = passwordChars[b[i]/5]
+	}
+	return b, nil
+}
+
+func (t *TSDBSecretsInfo) createTimescaleDBCredentials() error {
+	secretName := t.ReleaseName + "-credentials"
+	exists, err := k8s.CheckSecretExists(secretName, t.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		fmt.Printf("using existing %s secret\n", secretName)
+		return nil
+	}
+
+	fmt.Printf("Creating TimescaleDB %s secret\n", secretName)
+	repPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	adminPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	superUserPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	sec := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: t.Namespace,
+		},
+		Data: map[string][]byte{
+			"PATRONI_REPLICATION_PASSWORD": repPass,
+			"PATRONI_admin_PASSWORD":       adminPass,
+			"PATRONI_SUPERUSER_PASSWORD":   superUserPass,
+		},
+		Type: "Opaque",
+	}
+
+	return k8s.CreateSecret(sec)
+}
+
+func generateCerts() ([]byte, []byte, error) {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate rsa key: %v", err)
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate serial number: %v", err)
+	}
+
+	template := x509.Certificate{
+		SerialNumber:          serialNumber,
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(time.Hour * 24 * 1000),
+		KeyUsage:              x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &priv.PublicKey, priv)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create certificate: %v", err)
+	}
+
+	certOut := new(bytes.Buffer)
+	if err := pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: derBytes}); err != nil {
+		return nil, nil, fmt.Errorf("failed to write data to cert.pem: %v", err)
+	}
+
+	privBytes, err := x509.MarshalECPrivateKey(priv)
```

Comment:
can we copy over  `PEMEncodePrivateKey(privateKey *ecdsa.PrivateKey) ([]byte, error)` and use that?